File: DS/Array/CPP/A4.cpp
#include <bits/stdc++.h>

using namespace std;

vector <int> leftRotation(vector <int> a, int d) {
    // Complete this function
    vector<int> tempV;
    for(int i=0;i<d;i++) {
        tempV.push_back(a.at(i));
    }
    for(int i=0;i<a.size()-d;i++) {
        a[i]=a.at(i+d);
    }
    int count = 0;
    
    for(int i=a.size()-d;i<a.size();i++) {
        a[i]=tempV.at(count);
        count++;
    }
    
    return a;
}

/*

// Exceeded Time Limit

vector <int> leftRotation(vector <int> a, int d) {
    // Complete this function
    for(int i=0;i<d;i++) {
        int temp = a.at(0);
        for(int j=0;j<a.size()-1;j++) {
            a[j]=a.at(j+1);
        }
        a[a.size()-1]=temp;
    }
    return a;
}

*/

int main() {
    int n;
    int d;
    cin >> n >> d;
    vector<int> a(n);
    for(int a_i = 0; a_i < n; a_i++){
       cin >> a[a_i];
    }
    vector <int> result = leftRotation(a, d);
    for (ssize_t i = 0; i < result.size(); i++) {
        cout << result[i] << (i != result.size() - 1 ? " " : "");
    }
    cout << endl;


    return 0;
}

File: DS/Array/JS/A4.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function leftRotation(a, d) {
    // Complete this function
    var tempV = [];
    for(var i=0;i<d;i++) {
        tempV.push(a[i]);
    }
    for(var i=0;i<a.length-d;i++) {
        a[i]=a[i+d];
    }
    var count = 0;
    
    for(var i=a.length-d;i<a.length;i++) {
        a[i]=tempV[count];
        count++;
    }
    
    return a;
}

function main() {
    var n_temp = readLine().split(' ');
    var n = parseInt(n_temp[0]);
    var d = parseInt(n_temp[1]);
    a = readLine().split(' ');
    a = a.map(Number);
    var result = leftRotation(a, d);
    console.log(result.join(" "));



}
